tsdb: cannot Resize exemplars from 0 to non-0 value

**What did you do?**
Initialized TSDB with maxExemplars=0, then changed it to 100000

**What did you expect to see?**
No crash

**What did you see instead? Under which circumstances?**

```
panic: runtime error: integer divide by zero

goroutine 240 [running]:
github.com/prometheus/prometheus/tsdb.(*CircularExemplarStorage).Resize(0xc000223090, 0x61a8, 0x0)
	github.com/prometheus/prometheus/tsdb/exemplar.go:287 +0x2b6
github.com/prometheus/prometheus/tsdb.(*Head).ApplyConfig(0xc000d1ae00, 0xc000561bf0, 0xa, 0xc0005ddb00)
	github.com/prometheus/prometheus/tsdb/head.go:680 +0xa5
```

This line does `% len(oldBuffer)` which will crash if the old buffer is of size zero.
https://github.com/prometheus/prometheus/blob/6a31b28ca95c1da441c516709dc07f975bde1ba4/tsdb/exemplar.go#L287

I believe zero size is valid because of this comment:
https://github.com/prometheus/prometheus/blob/6a31b28ca95c1da441c516709dc07f975bde1ba4/tsdb/exemplar.go#L255

--- a/tsdb/exemplar.go
+++ b/tsdb/exemplar.go
@@ -279,7 +279,7 @@ func (ce *CircularExemplarStorage) Resize(l int64) int {
 
 	migrated := 0
 
-	if l > 0 {
+	if l > 0 && len(oldBuffer) > 0 {
 		// Rewind previous next index by count with wrap-around.
 		// This math is essentially looking at nextIndex, where we would write the next exemplar to,
 		// and find the index in the old exemplar buffer that we should start migrating exemplars from.

--- a/tsdb/exemplar_test.go
+++ b/tsdb/exemplar_test.go
@@ -413,7 +413,7 @@ func TestResize(t *testing.T) {
 			expectedMigrated:  50,
 		},
 		{
-			name:              "Zero",
+			name:              "ShrinkToZero",
 			startSize:         100,
 			newCount:          0,
 			expectedSeries:    []int{},
@@ -432,6 +432,14 @@ func TestResize(t *testing.T) {
 			name:              "NegativeToNegative",
 			startSize:         -1,
 			newCount:          -2,
+			expectedSeries:    []int{},
+			notExpectedSeries: []int{},
+			expectedMigrated:  0,
+		},
+		{
+			name:              "GrowFromZero",
+			startSize:         0,
+			newCount:          10,
 			expectedSeries:    []int{},
 			notExpectedSeries: []int{},
 			expectedMigrated:  0,

